lib/reline/config.rb

```diff
@@ -226,7 +227,12 @@ def handle_directive(directive, file, no, if_stack)
     when 'if'
       condition = false
       case args
-      when 'mode'
+      when /^mode=(vi|emacs)$/i
+        mode = $1.downcase
+        if @editing_mode_label == mode.to_sym
+          condition = true
+          @keymap_label = mode == 'emacs' ? :emacs : :vi_insert
```

Comment:
I think `if mode=xxx` does not modify keymap_label.

I tried this inputrc
```inputrc
set editing-mode vi
set keymap emacs
$if mode=vi
  "x": "FOOBAR"
$endif
```
With this command. Typing x in vi mode just inserts x. After switching to emacs mode with \e\C-e, typing x inserts FOOBAR.
```
$ ruby -e 'puts "vi:x, \e\C-e\C-eemacs:x"' | ruby -rreadline -e "puts Readline::VERSION;Readline.readline('>')"
8.2
>vi:x, emacs:FOOBAR
```


I think Readline reads inputrc like this
```inputrc
# Sets mode_label and keymap_label to vi
set editing-mode vi
# Change keymap_label to emacs. mode_label is still vi.
set keymap emacs
# condition=true because current mode_label is vi
$if mode=vi
  # sets keybinding to current keymap_label=emacs
  "x": "FOOBAR"
$endif
```

Reply:
Thanks for letting me know. I fixed. If you see anything else wrong, please let me know 🙏